core/controllers/topic_viewer.py

```diff
@@ -34,51 +55,30 @@ def get(self, topic_name):
             raise self.PageNotFoundException
 
         topic = topic_services.get_topic_by_name(topic_name)
-        canonical_stories = [
-            story_services.get_story_by_id(
+        canonical_story_summaries = [
+            story_services.get_story_summary_by_id(
                 canonical_story_id) for canonical_story_id
             in topic.canonical_story_ids]
-        additional_stories = [
-            story_services.get_story_by_id(
+        additional_story_summaries = [
+            story_services.get_story_summary_by_id(
                 additional_story_id) for additional_story_id
             in topic.additional_story_ids]
 
         canonical_story_dicts = [{
             'id': story.id,
             'title': story.title,
             'description': story.description
-        } for story in canonical_stories]
+        } for story in canonical_story_summaries]
```

Comment:
Dummy variable should be "summary". Should also still call summary.to_dict() or a new, similar, method. Encapsulate logic in the logic layer and add unit tests for it.

Ditto below. I think I mentioned this in a previous review comment.

Reply:
Sorry, I don't understand. Why do we need to do summary.to_dict() We need just these three values for now.

Reply:
Generally speaking, you should not be manually composing things in the controller layer. I am talking about moving the logic to a method on the story summary object.

It could be to_dict() or a separate method that is appropriately named. But the controller should just be doing `canonical_story_dicts= [summary.to_dict() for summary in canonical_story_summaries]` (or, if you're defining a new function, then summary.{{new_function_name}}()).

Reply:
Okay got it! I've defined a new method I summary_services as is used for creator_dashboard to get summary dicts. Hope this is fine.